core/core-exchange/src/main/java/ai/wanaku/core/exchange/ParsedToolInvokeRequest.java

```diff
@@ -41,8 +42,13 @@ public static ParsedToolInvokeRequest parseRequest(String uri, ToolInvokeRequest
 
         String parsedUri = URIParser.parse(uri, map);
 
+        //Add additional configuration
+        parsedUri = URIHelper.addQueryParameters(parsedUri, toolInvokeRequest.getServiceConfigurationsMap());
+
```

Comment:
So, I'm thinking we should have this as an optional post processing step. I believe that not all services will want to have to append service configurations to the URI (specially, those that are not Camel-based). So, I think we should have a `Function<String, String> postProcessor` as an argument, that can receives the URI, does something like `parsedUri = postProcessor.accept(parsedUri)`. This would give more flexibility for those services ... 

To be clear: I don't think you need to change this now. I'm just dumping a thought that came to me while reviewing this ...

Reply:
@orpiske I like the idea of post processor functions, but we cannot pass it in the configuration, it should be provided by the service. There should be a sort of trigger that fires the post processor execution functions, in the case above, something like  predicate: 

`if target == 'configuration' and scope == 'service-endpoint'`


WDYT?

Reply:
Yeah, sounds good to me. 

Obs.: we can leave this for later ... I'm mostly thinking about a future improvement (and also the fact that we need to consider non-Camel workloads).